app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -210,9 +212,28 @@ class HomeFragment : Fragment() {
         val activity = activity as HomeActivity
         activity.themeManager.applyStatusBarTheme(activity)
 
+        setFragmentFullScreen(this.activity?.window, view.homeLayout)
+
         return view
     }
 
+    private fun setFragmentFullScreen(window: Window?, homeScreen: View) {
```

Comment:
I'd prefer to extract this into a separate `LifecycleObserver` (with onCreate, onResume, onPause methods) so we can reuse it elsewhere and make it more testable.

Reply:
The layout adjustment has to be made in `onCreateView` not in `onCreate`. IDK if I can implement a `LifecycleObserver` for this.

Reply:
`Fragment#viewLifecycleOwner`'s create event corresponds to `onCreateView`, so a `LifecycleObserver` can be used here.

Reply:
IMO it is a bit of overkill  since I do not think we will use it on other fragments. But if you feel strongly about this implementation, I will try to get something working tomorrow. TY for your input, it's always welcomed and valued!

Reply:
I feel strongly about it since `HomeFragment` is pretty enormous and hard to test, so I'm trying to separate out whatever I can. If you like we can land this for now and make that change later :)

Reply:
You're right about this! I agree, we can land this and I will get to the change next week! Thanks for suggesting this!